extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1000,20 +1006,43 @@ public String getHost() {
 
     @Nonnull
     @Override
-    public String getPrivacy() {
-        return "";
+    public Privacy getPrivacy() {
+        boolean isUnlisted = playerResponse
+                .getObject("microformat")
+                .getObject("playerMicroformatRenderer")
+                .getBoolean("isUnlisted");
+        return isUnlisted ? Privacy.UNLISTED : Privacy.PUBLIC;
     }
 
     @Nonnull
     @Override
     public String getCategory() {
-        return "";
+        try {
+            return playerResponse.getObject("microformat")
+                    .getObject("playerMicroformatRenderer")
+                    .getString("category");
+        } catch (Exception ignored) {
+            return "";
+        }
     }
 
     @Nonnull
     @Override
     public String getLicence() {
-        return "";
+        try {
+            JsonArray rows = initialData.getObject("contents")
+                    .getObject("twoColumnWatchNextResults").getObject("results").getObject("results")
+                    .getArray("contents").getObject(1).getObject("videoSecondaryInfoRenderer")
+                    .getObject("metadataRowContainer").getObject("metadataRowContainerRenderer").getArray("rows");
+            JsonObject metadataRowRenderer = rows.getObject(rows.size() - 1) // it is usually last
+                    .getObject("metadataRowRenderer");
+            String probablyLicence = getTextFromObject(metadataRowRenderer.getArray("contents").getObject(0));
+            return getTextFromObject(metadataRowRenderer.getObject("title")).toLowerCase().contains("licensed")
+                    ? probablyLicence
+                    : "";
+        } catch (Exception e) {
+            return "";
+        }
     }
```

Comment:
This doesn't actually get the license, but the licensor. See e.g. https://www.youtube.com/watch?v=IuArYvVbUo0 for a video that has listed its license. Also: `final`.

Reply:
Ok, great. And for the licence, can we have something dynamic? 

I mean Peertube has only 7 licences: https://peertube.cpy.re/api/v1/videos/licences, then it'll make it easier to translate to have them dynamic, and the rest is untranslated.
For YouTube, it seems like it's either Youtube licence (default) or a CC-BY (attribution).
https://www.youtube.com/static?template=terms (it states default licence is YouTube one).
https://support.google.com/youtube/answer/2797468?hl=en (it states we can choose CC by, i.e attribution, but can we have other CC licences?).
Then for YouTube: if no licence is shown, it means YouTube licence. Otherwise for CC-BY, we can use the same field in enum as peertube.
And for Soundcloud, I didn't really understand, there were several field, like C and P copyright (you should just download a track and look at the json).

Reply:
On SoundCloud, there's the `license` field containing e.g. `cc-by`, see e.g. https://soundcloud.com/wearecc/plays-well-with-others-ep-2-what-do-an-army-of-ants-and-an-online-encyclopedia-have-in-common:

```json
{
  "comment_count": 7,
  "full_duration": 1400920,
  "downloadable": false,
  "created_at": "2019-03-28T13:36:18Z",
  "description": "“Stigmergy is a mechanism of indirect coordination, through the environment, between agents or actions,” according to Wikipedia. “The principle is that the trace left in the environment by an action stimulates the performance of a next action, by the same or a different agent.” Wikipedia should know all about stigmergy, since it’s almost a perfect description of how Wikipedia itself is made. While this model of collaboration is most commonly associated with ants, not global knowledge sharing, we’ll explore both and how they’ve shaped our world through asynchronous collaboration.\n\nJoin us as we talk to Katherine Maher, the executive director of the Wikimedia Foundation (the nonprofit organization that runs Wikipedia), and Clint Penick, an ant researcher and assistant research professor in the Biomimicry Center at Arizona State University.\n\n===\n\nAll original content in Plays Well with Others is available under a Creative Commons BY license.\n\nThis episode contains music by Blue Dot Sessions, available under a Creative Commons BY-NC license at the Free Music Archive.\n\nWe also use sound effects, available under CC licenses and in the public domain, from Freesound.\n\nFor links to these tracks and for more information about our show, please visit playswellwithothers.org.",
  "media": {
    "transcodings": [
      {
        "url": "https://api-v2.soundcloud.com/media/soundcloud:tracks:597253485/dac2dc88-1b30-49aa-8f11-4176901591ec/stream/hls",
        "preset": "mp3_0_0",
        "duration": 1400920,
        "snipped": false,
        "format": {
          "protocol": "hls",
          "mime_type": "audio/mpeg"
        },
        "quality": "sq"
      },
      {
        "url": "https://api-v2.soundcloud.com/media/soundcloud:tracks:597253485/dac2dc88-1b30-49aa-8f11-4176901591ec/stream/progressive",
        "preset": "mp3_0_0",
        "duration": 1400920,
        "snipped": false,
        "format": {
          "protocol": "progressive",
          "mime_type": "audio/mpeg"
        },
        "quality": "sq"
      },
      {
        "url": "https://api-v2.soundcloud.com/media/soundcloud:tracks:597253485/452f1e05-f425-451f-a257-858c039c55b9/stream/hls",
        "preset": "opus_0_0",
        "duration": 1400743,
        "snipped": false,
        "format": {
          "protocol": "hls",
          "mime_type": "audio/ogg; codecs=\"opus\""
        },
        "quality": "sq"
      }
    ]
  },
  "title": "Plays Well with Others, Ep 2: What Do an Army of Ants and an Online Encyclopedia Have in Common?",
  "publisher_metadata": {
    "urn": "soundcloud:tracks:597253485",
    "contains_music": true,
    "id": 597253485
  },
  "duration": 1400920,
  "has_downloads_left": true,
  "artwork_url": "https://i1.sndcdn.com/artworks-000511451328-51jljl-large.jpg",
  "public": true,
  "streamable": true,
  "tag_list": "storytelling collaboration wikipedia ants stigmergy \"creative commons\"",
  "genre": "Podcast",
  "id": 597253485,
  "reposts_count": 6,
  "state": "finished",
  "label_name": null,
  "last_modified": "2019-09-06T21:39:16Z",
  "commentable": true,
  "policy": "ALLOW",
  "visuals": null,
  "kind": "track",
  "purchase_url": null,
  "sharing": "public",
  "uri": "https://api.soundcloud.com/tracks/597253485",
  "secret_token": null,
  "download_count": 0,
  "likes_count": 23,
  "urn": "soundcloud:tracks:597253485",
  "license": "cc-by",
  "purchase_title": null,
  "display_date": "2019-03-28T13:36:18Z",
  "embeddable_by": "all",
  "release_date": null,
  "user_id": 1457130,
  "monetization_model": "NOT_APPLICABLE",
  "waveform_url": "https://wave.sndcdn.com/bKOA7Pwbut93_m.json",
  "permalink": "plays-well-with-others-ep-2-what-do-an-army-of-ants-and-an-online-encyclopedia-have-in-common",
  "permalink_url": "https://soundcloud.com/wearecc/plays-well-with-others-ep-2-what-do-an-army-of-ants-and-an-online-encyclopedia-have-in-common",
  "user": {
    "avatar_url": "https://i1.sndcdn.com/avatars-000001558477-l96tko-large.jpg",
    "first_name": "Creative Commons",
    "full_name": "Creative Commons",
    "id": 1457130,
    "kind": "user",
    "last_modified": "2019-03-09T22:03:03Z",
    "last_name": "",
    "permalink": "wearecc",
    "permalink_url": "https://soundcloud.com/wearecc",
    "uri": "https://api.soundcloud.com/users/1457130",
    "urn": "soundcloud:users:1457130",
    "username": "Creative Commons",
    "verified": false,
    "city": "Internet",
    "country_code": null
  },
  "playback_count": 19946
}
```

I'm not sure what the possible options are for YouTube or SoundCloud though.